sandstone/client/sandstone/components/filetreedirective/tests/filetree.directive.test.js

```diff
@@ -1,208 +1,421 @@
+              expect(isolateScope.treeData.filetreeContents.length).toBe(2);
+          });
+          it('getDirContents', function(){
+              isolateScope.getDirContents(isolateScope.treeData.filetreeContents[0], true);
+              // First directory should have 4 files
+              expect(isolateScope.treeData.filetreeContents[0].children.length).toBe(4);
+          });
+      });
+      describe('template', function() {
+          beforeEach(function() {
+              scope.sd = {
+                  noSelections: true,
+                  multipleSelections: true,
+                  dirSelection: false
+              };
+              var el = angular.element('<div sandstone-filetree tree-data="ctrl.treeData" leaf-level="dir" selection-desc="sd"></div>');
+              element = $compile(el)(scope);
+              scope.$digest();
+
+              // Get isolate scope
+              isolateScope = element.isolateScope();
+          });
+          var getExpandedNodes = function(tpl) {
+              var matches = tpl.match(/fa-caret-down/g);
+              if(!matches) {
+                  return [];
+              }
+              return matches;
+          };
+
+          var getContractedNodes = function(tpl) {
+              var matches = tpl.match(/fa-caret-right/g);
+              if(!matches) {
+                  return [];
+              }
+              return matches;
+          };
+
+          var getSelectedNodes = function(tpl) {
+              var matches = tpl.match(/tree-selected/g);
+              if(!matches) {
+                  return [];
+              }
+              return matches;
+          };
+
+          var getTemplate = function() {
+              scope.$digest();
+              return element.html();
+          };
+
+          describe('multipleSelections: false', function() {
+              it('should show correctly expanded nodes and leaf nodes', function() {
+                  // Expand the first node
+                  isolateScope.treeData.expandedNodes = [isolateScope.treeData.filetreeContents[0]];
+                  isolateScope.getDirContents(isolateScope.treeData.filetreeContents[0], true);
+
+                  var template = getTemplate();
+                  var expandedNodes = getExpandedNodes(template);
+                  var contractedNodes = getContractedNodes(template);
+                  expect(expandedNodes.length).toBe(1);
+                  expect(contractedNodes.length).toBe(5);
+              });
+              it('should have a single selected node', function() {
+                  // Select the first node
+                  isolateScope.describeSelection(isolateScope.treeData.filetreeContents[0], true);
+                  var template = getTemplate();
+                  var selectedNodes = getSelectedNodes(template);
+                  expect(selectedNodes.length).toBe(1);
+              });
+          });
+
+          describe('multipleSelections: false', function() {
```

Comment:
Multiselection can be toggled by changing `$scope.sd` on an already-compiled directive, so creating a nested `describe` block with new compilation code is unnecessary.